### PR TITLE
Set Site Domains Screen title and right bar button using UIKit

### PR DESCRIPTION
**Pull request by salimbraksa**
Refs https://github.com/wordpress-mobile/WordPress-iOS/issues/21777

## Description

This PR fixes some navigation bar UI glitches during the "My Site > Site Domains > All Domains" navigation.

| Before | After |
| ------ | ------ |
| <video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/bdef7e23-cba5-49db-9758-fac82a49b276"></video> |  <video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/2fbba126-6eba-40ee-9e1e-4d2489431c7b"></video> |

## Test Instructions
Same as https://github.com/wordpress-mobile/WordPress-iOS/pull/21973

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Domains/Views/SiteDomainsView.swift [hunk 1]
```diff
@@ -2,7 +2,8 @@ import SwiftUI
 import WordPressKit
 
 /// The Domains dashboard screen, accessible from My Site
-struct DomainsDashboardView: View {
+struct SiteDomainsView: View {
```
Comment (hassaanelgarem) [inline WordPress/Classes/ViewRelated/Domains/Views/SiteDomainsView.swift:5]: Thanks for renaming this! I kept confusing it as a "Dashboard" related component.

**Review by hassaanelgarem — COMMENTED** [review]
I noticed that the "My Domains" CTA is visible even if the feature flag is disabled.

Other than that, it works as expected 🚀

**Comment by salimbraksa** [comment]
> I noticed that the "My Domains" CTA is visible even if the feature flag is disabled.

Resolved in 38b766ef431b810847ed7e8e2d134c188c019da5

**Review by hassaanelgarem — APPROVED** [review]
LGTM! 🚀 

The CI is failing though, looks like a build error on WordPress.

**Review by hassaanelgarem — APPROVED** [review]

**Pull request merged**